### ISSUE TITLE
Request: sharing `opt_set_local` feedback

### ISSUE THREAD

**Issue by dgkf** [dataheld/elf#5]
Hi @maxheld83! Very happy to see `options` getting adopted!

I noticed that you made your own `withr` flavor of `opt_set_local` and I was hoping you could share your experience here.  I've been hesitating about exporting this function because I don't fully understand the edge cases that `withr` addresses that are not handled via `on.exit`. Is it primarily to safeguard against a future `on.exit(.., add = FALSE)`?

If you have explicit edge cases that require the use of `withr`, I'd love to hear about them so that I can make a more informed decision about what an exported version of this function should look like.

**Comment by maxheld83** [comment]
thanks again so much for the amazing options @dgkf!
I use it all the time. 

My TL,DR is: I use *withr* for it's syntactic sugar, and would love either a *withr*-based `options::opt_set_local()`, or even `options::with_opt()`/`options::local_opt()` (-> https://github.com/dgkf/options/issues/18).

Long version:

I just use *withr* for laziness.
I've gotten so used to its semantics, that I'm now afraid of the extra thinking I'd have to do, with I use plain `on.exit()` instead.

Specifically:

1. I like the message/helpers which `withr::defer()` gives me if I set it on the global environment, which I often do during interactive testing, dev.
  I've found this to be a tremendous footgun and thus appreciate the protections against it from `withr::defer()`:
  
  ```
  Setting global deferred event(s).
i These will be run:
  * Automatically, when the R session ends.
  * On demand, if you call `withr::deferred_run()`.
i Use `withr::deferred_clear()` to clear them without executing.
   ```
2. I like to use both `withr::local_()`/`withr::with_()`, whichever is easiest in the context. Easily possible of course with `on.exit()`, but somewhat less expressive.
3. As you point out and warn in the docs, I do indeed almost always want to use `on.exit(add = TRUE, after = FALSE)`, but enjoy not having to think about it.
